### PR TITLE
Add PAC cel filtering propal

### PR DESCRIPTION
**Pull request by chmouel**
Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

### CODE REVIEW

File: enhancements/pac-cel-filtering.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+title: Option to disable default creation of service account and rbac resources
```
Comment (sm43) [inline enhancements/pac-cel-filtering.md:2]:
```suggestion
title: Pipelines as code event filtering with CEL
```
😄

File: enhancements/pac-cel-filtering.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+We would still keep the basic filtering mechanism for now and only let the user use the `on-cel-expression` advanced for advance usage.
+
+## Advantage
+
+- We can address most of the Gitlab action use cases <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on>
+- Use should be familiar with CEL syntax if they are familiar with triggers.
+
+## Cons
+
+- are we reinventing a `"trigger as code"` ? 🙃
```
Comment (vdemeester) [inline enhancements/pac-cel-filtering.md:68]: Maybe ? 😝

File: enhancements/pac-cel-filtering.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+# TEP-XXX: Pipelines as code event filtering with CEL
+
+## Summary
+
+This proposal helps user to do advanced filtering for PipelineRun selections.
+
+## Motivation
+
+We currently have a very basic filtering mechanism for Pac PipelineRuns.
```
Comment (sm43) [inline enhancements/pac-cel-filtering.md:17]: you mean like fetching the actual user pipelineRun by labels right?
using kubectl
with pac labels

**Comment by chmouel** [comment]
Here is a proof of concept. The change in the annotation matcher : 

https://github.com/chmouel/pipelines-as-code/commit/66b2438786dfe2e45f414b4142ab9d3af830dbcc#diff-7e21d71dc0c56798fde36ab4207d45899ac473d6287b63a93a7cbd5c0d92e484R104

The new pipelinerun that only match on a  pull_request event, target_branch main and source_branch cel_event

https://github.com/chmouel/scratchmyback/blob/9caea22/.tekton/run.yaml#L6-L7

and pipeline matches it : 

image

I think that looks pretty nice,

File: enhancements/pac-cel-filtering.md (hunk 2 above)
Comment (rupalibehera) [reply enhancements/pac-cel-filtering.md:68]: looks like stuck_out_tongue_closed_eyes

File: enhancements/pac-cel-filtering.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Using the `pipelinesascode.tekton.dev/on-*` annotations user can specify wether it
+targets a specific heads like branch or tags or an event (pull_request, issue).
+
+```yaml
+    pipelinesascode.tekton.dev/on-target-branch: "[%s]"
+    pipelinesascode.tekton.dev/on-event: "[%s]"
+```
+
+This works for most cases but thi misses some flexibility for some advanced use cases.
```
Comment (rupalibehera) [inline enhancements/pac-cel-filtering.md:27]: nitpick
```suggestion
This works for most cases but this misses some flexibility for some advanced use cases.
```

File: enhancements/pac-cel-filtering.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+```
+
+we can imagine any filters that can be expressed in CEL we want to grow with, including some custom function like triggers has.
+
+We would still keep the basic filtering mechanism for now and only let the user use the `on-cel-expression` advanced for advance usage.
+
+## Advantage
+
+- We can address most of the Gitlab action use cases <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on>
+- Use should be familiar with CEL syntax if they are familiar with triggers.
```
Comment (rupalibehera) [inline enhancements/pac-cel-filtering.md:64]: nitpick
```suggestion
- User should be familiar with CEL syntax if they are familiar with triggers.
```

File: enhancements/pac-cel-filtering.md (hunk 2 above)
Comment (chmouel) [reply enhancements/pac-cel-filtering.md:68]: triggers doesn't do "as code", we are using triggers atm  and planning to get away from it..

**Review by vdemeester — APPROVED** [review]

**Comment by chmouel** [comment]
Thanks for the reviews and from @siamaksade thumb up as well it seems this can get implemented. 

this will open a world of possibilities for flexible pipeline and advanced matching with pac

**Pull request merged**